mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala

```diff
@@ -566,26 +565,52 @@ object LocalLDAModel extends MLReadable[LocalLDAModel] {
     }
   }
 
+  private case class Data(
+                           vocabSize: Int,
+                           topicsMatrix: Matrix,
+                           docConcentration: Vector,
+                           topicConcentration: Double,
+                           gammaShape: Double)
+
   private class LocalLDAModelReader extends MLReader[LocalLDAModel] {
 
     private val className = classOf[LocalLDAModel].getName
 
     override def load(path: String): LocalLDAModel = {
+      // Import implicits for Dataset Encoder
+      val sparkSession = super.sparkSession
+      import sparkSession.implicits._
+
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
       val data = sparkSession.read.parquet(dataPath)
-        .select("vocabSize", "topicsMatrix", "docConcentration", "topicConcentration",
-          "gammaShape")
-        .head()
-      val vocabSize = data.getAs[Int](0)
-      val topicsMatrix = data.getAs[Matrix](1)
-      val docConcentration = data.getAs[Vector](2)
-      val topicConcentration = data.getAs[Double](3)
-      val gammaShape = data.getAs[Double](4)
+      val vectorConverted = MLUtils.convertVectorColumnsToML(data, "docConcentration")
+      val Row(vocabSize: Int, topicsMatrix: Matrix, docConcentration: Vector,
+        topicConcentration: Double, gammaShape: Double) = MLUtils.convertMatrixColumnsToML(
+        vectorConverted, "topicsMatrix").as[Data]
       val oldModel = new OldLocalLDAModel(topicsMatrix, docConcentration, topicConcentration,
         gammaShape)
       val model = new LocalLDAModel(metadata.uid, vocabSize, oldModel, sparkSession)
-      DefaultParamsReader.getAndSetParams(model, metadata)
+
+      metadata.sparkVersion match {
+        case "1.6" =>
+          implicit val format = DefaultFormats
+          metadata.params match {
+           case JObject(pairs) =>
+             pairs.foreach { case (paramName, jsonValue) =>
+               val origParam =
+                 if (paramName == "topicDistribution") "topicDistributionCol" else paramName
+               val param = model.getParam(origParam)
+               val value = param.jsonDecode(compact(render(jsonValue)))
+               model.set(param, value)
+             }
+           case _ =>
+             throw new IllegalArgumentException(
+               s"Cannot recognize JSON metadata: ${metadata.metadataJson}.")
+           }
+        case "2.x" =>
```

Comment:
I don't think you can match on "2.x"  You'll need a more robust comparison of versions.  (Check out what sparkSession.version returns; it could be things like "2.0.1" or "2.0.1-SNAPSHOT".)